### ISSUE TITLE
*ngSwitchDefault Odd Behavior

### ISSUE THREAD

**Issue by mootinator** [angular/angular#9420]
**I'm submitting a ...**  (check one with "x")

```
[ x ] bug report
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
*ngSwitchDefault appears to create a component with an unusual lifecycle even when one of the other cases is matched.

**Expected/desired behavior**

I would expect it to not do that.

**Reproduction of the problem**
If the current behavior is a bug or you can illustrate your feature request better with an example, please provide the steps to reproduce and if possible a minimal demo of the problem via https://plnkr.co or similar (you can use this template as a starting point: http://plnkr.co/edit/tpl:AvJOMERrnz94ekVua0u5).

http://plnkr.co/edit/bO1pI7COc5rYPENwqeLH

Here is the console output after clicking the button on the plunkr several times:

destroy undefined
init 0
destroy 0
destroy undefined
init 1
destroy 1
destroy undefined
init 2
destroy 2
destroy undefined
init 3
destroy 3
destroy undefined
init 4
destroy 4

**What is the expected behavior?**

The output should not contain the 'destroy undefined' lines, and they're not there if *ngSwitchDefault is not used.

**What is the motivation / use case for changing the behavior?**

Possible memory leak?

**Please tell us about your environment:**
- **Angular version:** 2.0.0-rc.2
- **Browser:** Chrome
- **Language:** TypeScript

**Comment by vicb** [comment]
Please make a minimal repro with only the required code to exercise the pb & give steps to reproduce. Thanks

**Comment by mootinator** [comment]
Minimal plunkr:
http://plnkr.co/edit/0bsUOV8gcwdD0E3bYlB6

To reproduce: 
Activate part of the template containing an [ngSwitch] block.
Both *ngSwitchCase(s) and *ngSwitchDefault must be present.
One of the cases must match.

**Comment by pkozlowski-opensource** [comment]
Here is a simplified plunk: http://plnkr.co/edit/myu6F8mBClhFaVIpMxxz?p=preview

tl;dr; works as expected

I think that you are misinterpreting how `ngSwitchDefault` works. As explained in https://angular.io/docs/ts/latest/api/common/index/NgSwitchDefault-directive.html it:

> Default case statements are displayed when **no match expression** matches the switch expression value.

**Comment by mootinator** [comment]
That simplified plunk is _completely_ missing the point.

If you have a ngSwitch inside an if block, and specifically the whole switch goes from not visible to visible, and a case other than the default is displayed when it becomes visible, the ngOnDestroy of a non-instantiated version of the default is spuriously being called.

**Comment by pkozlowski-opensource** [comment]
Oh I see. Please, in the future provide a minimal plunker and crystal - clear description - it will avoiding this type of going back and for...
